### PR TITLE
Upgrade to RA v5

### PR DESCRIPTION
**Pull request by erwanMarmelab**
## Todo

- [x] Upgrade depedencies
- [x] Fix demo's build errors
- [x] Test demos

### CODE REVIEW

File: packages/demo-react-admin/package.json [hunk 1]
```diff
@@ -15,38 +14,30 @@
         "@emotion/styled": "^11.6.0",
         "@mui/icons-material": "^5.0.1",
         "@mui/material": "^5.0.2",
+        "@tanstack/react-query": "^5.8.4",
+        "@tanstack/react-query-devtools": "^5.8.4",
         "jsonexport": "^3.2.0",
         "lodash": "~4.17.5",
         "prop-types": "^15.7.2",
```
Comment (slax57) [inline packages/demo-react-admin/package.json]: Can we get rid of `prop-types`?

File: packages/demo-react-admin/package.json [hunk 2]
```diff
@@ -15,38 +14,30 @@
...
         "@hookform/devtools": "^4.0.2",
         "@types/jest": "^29.2.3",
         "@vitejs/plugin-react-refresh": "^1.3.3",
-        "html-loader": "~3.1.0",
-        "html-webpack-plugin": "~5.5.0",
-        "ignore-not-found-export-plugin": "^1.0.1",
-        "react-app-polyfill": "^1.0.4",
-        "style-loader": "~3.3.1",
-        "ts-loader": "^9.2.8",
         "typescript": "^4.4.0",
```
Comment (slax57) [inline packages/demo-react-admin/package.json]: `typescript` version should be upgraded here too

File: packages/demo-react-admin/package.json [hunk 3]
```diff
@@ -15,38 +14,30 @@
...
-        "vite": "^2.2.3",
-        "webpack": "~5.70.0",
-        "webpack-cli": "^4.10.0",
-        "webpack-dev-server": "~4.7.4"
+        "vite": "^2.2.3"
```
Comment (slax57) [inline packages/demo-react-admin/package.json]: `vite` version too

File: packages/ra-auth-auth0/package.json [hunk 4]
```diff
@@ -17,8 +17,8 @@
     "peerDependencies": {
         "@auth0/auth0-spa-js": "^2.0.0",
         "ra-core": "^4.10.0",
```
Comment (slax57) [inline packages/ra-auth-auth0/package.json]: Can you please replaces all direct dependencies to `ra-core` by dependencies to `react-admin`?

Also, you need to use v5.

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-auth-auth0/package.json (hunk 4 above)
Comment (erwanMarmelab) [reply packages/ra-auth-auth0/package.json]: I'm not sur we should do it. @fzaninotto change it in v1.1.0
cf https://github.com/marmelab/ra-auth-auth0/blob/main/CHANGELOG.md#110

Comment (slax57) [reply packages/ra-auth-auth0/package.json]: @fzaninotto only rewrote the changelog, but the actual change can be tracked back to a commit by @djhi : https://github.com/marmelab/ra-auth-auth0/commit/1595814616e96dc042f72af5c3e71d7023d92a8f#diff-34c62cae6b5d6d2d2e7a38bbb31cbd58180975ed57c7a158be7d8d3192e0ffa7

IMO we should still do it. I see no good reason to have `ra-core` and not `react-admin`, added to the fact that, [at the beginning](https://github.com/marmelab/ra-auth-auth0/blob/307909c59079a8400bc0107481907d510c4b9d8c/packages/ra-auth-auth0/package.json), we had both.

**Review by slax57 — CHANGES_REQUESTED** [review]
There are compilation errors in the demo.

You can run `yarn type-check` while in the `packages/demo-react-admin` folder to see them.

**Review by slax57 — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,9 @@
+## 2.0.0
+
+-   Upgrade `react-admin` to v5
+-   Remove prop-types
+-   Change imports from `ra-core` to `react-admin`
```
Comment (slax57) [inline CHANGELOG.md:5]:
```suggestion
-   Change imports from `ra-core` to `react-admin`
- [Doc] Fix missing import and enhance `httpClient` data provider section
```

**Pull request merged**